tests/fullstack-test/expr/regexp.test

```diff
@@ -71,13 +71,12 @@ mysql> set @@tidb_isolation_read_engines='tiflash'; set @@tidb_enforce_mpp=1; se
 
 mysql> drop table if exists test.t
 mysql> create table test.t (data varchar(30), pattern varchar(30)); 
-mysql> insert into test.t values ('', ''), ('abcd', 'abcd');
+mysql> insert into test.t values ('abcd', 'abcd');
 mysql> alter table test.t set tiflash replica 1
 func> wait_table test t
-mysql> set @@tidb_isolation_read_engines='tiflash'; set @@tidb_enforce_mpp=1; select data regexp pattern, data regexp '', '' regexp pattern from test.t;
-+---------------------+----------------+-------------------+
-| data regexp pattern | data regexp '' | '' regexp pattern |
-+---------------------+----------------+-------------------+
-|                   1 |              1 |                 1 |
-|                   1 |              1 |                 0 |
-+---------------------+----------------+-------------------+
+mysql> set @@tidb_isolation_read_engines='tiflash'; set @@tidb_enforce_mpp=1; select data regexp pattern, '' regexp pattern from test.t;
```

Comment:
Need add some tests to test the match type

Reply:
> Need add some tests to test the match type

I will do this